mxcubeweb/routes/login.py

```diff
@@ -45,15 +32,17 @@ def login():
         password = params.get("password", "")
 
         try:
-            res = jsonify(app.usermanager.login(login_id, password))
+            app.usermanager.login(login_id, password)
         except Exception as ex:
             msg = "[LOGIN] User %s could not login (%s)" % (
                 login_id,
                 str(ex),
             )
             logging.getLogger("MX3.HWR").exception("")
             logging.getLogger("MX3.HWR").info(msg)
-            res = deny_access("Could not authenticate")
+            res = make_response(jsonify({"msg": "Could not authenticate"}), 200)
+        else:
+            res = make_response(jsonify({"msg": ""}), 200)
 
         session.permanent = True
```

Comment:
I managed to get the tests working by adding back this line which I initially removed, as I thought that setting `SESSION_PERMANENT=True` would be enough and certainly as we are also using `SESSION_PERMANENT_LIFETIME` (and the later is taken into account). 

However it seems that the configuration value is not taken into account and we need to set it programmatically, have a look at our mockup server [config](https://github.com/mxcube/mxcubeweb/blob/b38bb1446733ced2ff3fc982aa6f386ee412b47d/test/HardwareObjectsMockup.xml/mxcube-web/server.yaml#L6)

Documentation even states that its true by default: https://flask-session.readthedocs.io/en/latest/config.html#configuration

Does someone have an idea of what that could be ?

Reply:
@marcus-oscarsson

I recall it not being straightforward to understand. It took me a while to wrap my head around sessions and their expiration. I had written [some doc](https://mxcubeweb.readthedocs.io/en/latest/dev/login.html), but obviously this doc is not good enough since reading it back now does not clarify much...

Flask and Flask-session have different behavior, different defaults. Flask does not seem to know about the `SESSION_PERMANENT` config setting at all.

I made a quick search and I am starting to winder if flask-session is used at all. I can not seem to find where it is used in the code. Am I missing something?

Reply:
Indeed that explains it, so looking at the [flask documentation](https://flask.palletsprojects.com/en/2.0.x/config/) `SESSION_PARMANENT` is not an option at all, however `SESSION_PERMANENT_LIFETIME` is and its used when one does `session.permanent=True`.  So I was simply confused by the Flask-session documentation, we were using it in the past but now we are using `Flask-Login`

Reply:
We should probably remove line 29 from `mxcubeweb/server.py`, line 6 from `test/HardwareObjectsMockup.xml/mxcube-web/server.yaml`, and finally line 20 from `mxcubeweb/core/models/configmodels.py`.

Reply:
:), Actually I thought if flask does not deal with the `SESSION_PERMANENT` setting we can :)

Reply:
On a second thought, thats just unnecessary ... Ill remove the config option as we are actually relying on this feature, it does not make any sense to set it to false in our case.